### PR TITLE
[4.0] Let build tools always start clean

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Clean all the Joomla core media folders each time that `npm install` is called

### Testing Instructions
Before applying the patch make a backup of your current media folder
Apply the patch
Run `npm install`
Compare the backup folder with the current media

### Documentation Changes Required

@wilsonge this can wait till 4.0 is released

### CODE REVIEW

**Comment by brianteeman** [comment]
Considering your previous comment is it safe to remove all of `vendor` and should it not be `vendor\xxxxx`

**Comment by dgrammatiko** [comment]
> Considering your previous comment is it safe to remove all of vendor and should it not be vendor\xxxxx

Yeah, probably I should target only the vendor folders that Joomla delivers. That said the existing code already discards the vendor folder, so I guess I need to fine-tune that part

File: build/build-modules-js/settings.json [hunk 1]
```diff
@@ -788,6 +788,67 @@
         "linkText": "Help me resolve this",
         "destFile": "/templates/system/fatal-error.html"
       }
-    }
+    },
+    "cleanUpFolders": [
```
Comment (richard67) [inline build/build-modules-js/settings.json:792]: Will someone remember to maintain this hardcoded list if some new extension which has assets will be added or removed in future?

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on a964dec2daabbee87e36737a9ffae2a5ccf58a65<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35048">issues.joomla.org/tracker/joomla-cms/35048</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on a964dec2daabbee87e36737a9ffae2a5ccf58a65<br /><br />Patch works. On loding the article the description gets truncated to 160 characters.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35048">issues.joomla.org/tracker/joomla-cms/35048</a>.</sub>

**Comment by richard67** [comment]
> I have tested this item ✅ successfully on [a964dec](https://github.com/joomla/joomla-cms/commit/a964dec2daabbee87e36737a9ffae2a5ccf58a65)
> 
> Patch works. On loding the article the description gets truncated to 160 characters.

@RickR2H Can it be that this second test here was meant for another PR? The comment below your test result doesn't fit to this PR here.

**Comment by RickR2H** [comment]
@richard67 My browser had some hick ups. Sorry for the confusion.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35048">issues.joomla.org/tracker/joomla-cms/35048</a>.</sub>

**Comment by RickR2H** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35048">issues.joomla.org/tracker/joomla-cms/35048</a>.</sub>

**Comment by RickR2H** [comment]
@dgrammatiko how can I verify that the patch works? Should there be a difference between the backed up folder and the newly generated?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35048">issues.joomla.org/tracker/joomla-cms/35048</a>.</sub>

**Comment by dgrammatiko** [comment]
> how can I verify that the patch works? Should there be a difference between the backed up folder

There should be identical (assuming that the backup didn't had `.gz` files, if it did you need to run the `npm run compress` iirc)

**Comment by dgrammatiko** [comment]
This PR is not needed urgently so I will switch it to draft and we can check it back after the GA

File: build/build-modules-js/settings.json (hunk 1 above)
Comment (dgrammatiko) [reply build/build-modules-js/settings.json:792]: This file is the backbone of the build tools but you're right there are way too many hardcoded things here. Will refactor it so it's easier for maintenance

**Comment by dgrammatiko** [comment]
@RickR2H @richard67 I've opened a new PR with the code from this one plus a replacement of a node package for the watch functionality. Please test https://github.com/joomla/joomla-cms/pull/35198

**Pull request closed without merging**